src/test/java/net/imglib2/algorithm/hough/HoughLineTransformTest.java

```diff
@@ -0,0 +1,171 @@
+
+	@Before
+	public void setUp() throws IOException
+	{
+		groundTruth = ImageIO.read( getClass().getResource( "result8.png" ) );
+		tpImg = ArrayImgs.unsignedBytes( 50l, 50l );
+		final RandomAccess< UnsignedByteType > randomAccess = tpImg.randomAccess();
+		for ( int i = 24; i < 25; ++i )
+		{
+			for ( int j = 14; j < 35; ++j )
+			{
+				randomAccess.setPosition( i, 0 );
+				randomAccess.setPosition( j, 1 );
+				randomAccess.get().set( 255 );
+			}
+		}
+	}
+
+	@Test
+	public < T extends RealType< T > > void testHoughLineTransformToTarget() throws IOException
+	{
+		// create votespace
+		final long[] dims = new long[ tpImg.numDimensions() ];
+		tpImg.dimensions( dims );
+		final long[] outputDims = HoughTransforms.getVotespaceSize( tpImg );
+		final Img< UnsignedByteType > votespace = new ArrayImgFactory().create( outputDims, tpImg.firstElement() );
+
+		// run transform
+		HoughTransforms.voteLines( tpImg, votespace );
+
+		// compare expected / actual dimensions
+		int height = groundTruth.getHeight();
+		int width = groundTruth.getWidth();
+		assertEquals( "Ground truth and result height do not match.", height, outputDims[ 1 ] );
+		assertEquals( "Ground truth and result width do not match.", width, outputDims[ 0 ] );
+
+		DataBufferByte db = ( DataBufferByte ) groundTruth.getData().getDataBuffer();
+		byte[] expected = db.getBankData()[ 0 ];
+
+		final RandomAccess< UnsignedByteType > ra = votespace.randomAccess();
+
+		// compare expected / actual results for regression.
+		int index = 0;
+		for ( int j = 0; j < height; ++j )
+		{
+			ra.setPosition( j, 1 );
+			for ( int i = 0; i < width; ++i )
+			{
+				ra.setPosition( i, 0 );
+
+				assertEquals( expected[ index++ ], ra.get().getRealDouble(), 0 );
+
+			}
+		}
+	}
+
+	@Test
+	public < T extends RealType< T > > void testPickPeaks()
+	{
+		// three expected peaks
+		final long[] expected = { 24, 89, 24, 90, 25, 91 };
+		int index = 0;
+
+		// create votespace
+		final long[] dims = new long[ tpImg.numDimensions() ];
+		tpImg.dimensions( dims );
+		final long[] outputDims = HoughTransforms.getVotespaceSize( tpImg );
+		final Img< UnsignedByteType > votespace = new ArrayImgFactory().create( outputDims, tpImg.firstElement() );
+
+		// run transform
+		HoughTransforms.voteLines( tpImg, votespace );
+
+		final UnsignedByteType minPeak = Util.getTypeFromInterval( votespace ).createVariable();
+		minPeak.setReal( 21 );
+
+		List< Point > peaks = HoughTransforms.pickPeaks( votespace, minPeak );
+		for ( Point p : peaks )
+		{
+			// assert dimension 0
+			assertEquals( expected[ index++ ], p.getLongPosition( 0 ), 0 );
+			// assert dimension 1
+			assertEquals( expected[ index++ ], p.getLongPosition( 1 ), 0 );
+		}
+	}
+
+	@Test
+	public void testEquationConversion()
+	{
+		final int rho = 2, theta = 45;
+		final double slope = HoughTransforms.getSlope( theta );
+		final double yint = HoughTransforms.getIntercept( rho, theta );
+
+		assertEquals( -1.0000000000000002, slope, 0 );
```

Comment:
Instead of using a high-precision floating point value here, pass a small delta, e.g. `1e-8` to `assertEquals`:
```
Assert.assertEquals( 1.0, slope, 1e-8 );
```
Please do the same for the assert on `yint`